app/assets/javascripts/govuk_publishing_components/components/modal-dialogue.js

```diff
@@ -2,14 +2,16 @@ window.GOVUK = window.GOVUK || {}
 window.GOVUK.Modules = window.GOVUK.Modules || {};
 
 (function (Modules) {
-  function ModalDialogue () { }
-
-  ModalDialogue.prototype.start = function ($module) {
-    this.$module = $module[0]
+  function ModalDialogue ($module) {
+    this.$module = $module
     this.$dialogBox = this.$module.querySelector('.gem-c-modal-dialogue__box')
     this.$closeButton = this.$module.querySelector('.gem-c-modal-dialogue__close-button')
     this.$html = document.querySelector('html')
     this.$body = document.querySelector('body')
+  }
+
+  ModalDialogue.prototype.init = function () {
+    if (!this.$dialogBox || !this.$closeButton) return
```

Comment:
I notice you've added this check in a few scripts. Can you explain why this is needed now?

Reply:
To make the scripts a bit more resilient. If the markup is not as expected (e.g. the script is called, but the component markup not, or has been altered by other scripts) we prevent the script from failing and potentially dragging the entire JavaScript execution off with it.